src/plays/github-user-search/GitHubUserSearch.jsx

```diff
@@ -0,0 +1,133 @@
+import {getPlayById} from "meta/play-meta-util";
+
+import PlayHeader from "common/playlists/PlayHeader";
+import "./gitHubUserSearch.css";
+import {useState} from "react";
+import axios from "axios";
+import {GitHubUserCard} from "./GitHubUserCard";
+
+function GitHubUserSearch(props) {
+  // Do not remove the below lines.
+  // The following code is to fetch the current play from the URL
+  const {id} = props;
+  const play = getPlayById(id);
+
+  // Your Code Start below.
+  const [query, setQuery] = useState("");
+  const [resData, setData] = useState({
+    data: null,
+    remaining_searches: 0,
+    time_until_reset: null,
+  });
+  const [dataFetchStates, setDataFetchStates] = useState({
+    loading: false,
+    error: false,
+  });
+
+  const getResetTime = (time) => {
+    const newTime = new Date(time * 1000);
+    return newTime.toLocaleTimeString();
+  };
+
+  const getRateLimitData = async () => {
+    const rate_limit = await axios.get("https://api.github.com/rate_limit");
+    const reset_time = getResetTime(rate_limit.data.resources.search.reset);
+    const reset_count = rate_limit.data.resources.search.remaining;
+    if (reset_count > 1)
+      setDataFetchStates((prev) => ({
+        ...prev,
+        error: false,
+      }));
+    setData((prev) => ({
+      ...prev,
+      remaining_searches: reset_count,
+      time_until_reset: reset_time,
+    }));
+  };
+
+  const getUsersData = async () => {
+    const res = await axios.get(
+      `https://api.github.com/search/users?q=${query}&per_page=15`
```

Comment:
what about the list of more than 15 results? Either pagination needs to be implemented or some textual information should be given on UI. In fact, you can create a number box for asking `response_size`

Reply:
Yes. Noted this. I feel Pagination is a indermediate concept and as this is a beginner play, I hard coded a fixed value. But yes, I can add the `response_size` input.